File: Second_Year/Semester 2/MPP/A3/A3/src/main/java/com/example/A2/domain/Product.java
package com.example.A2.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Cascade;

import java.util.List;


@Entity
@Table(name="product")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
public class Product {
    private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "product_id") Long id;

    private String name;

    private double price;

    private String description;

    private String color;

    private String category;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "product")
    private List<Review> reviewsList;

}

File: Third_Year/Semester1/Parallel_and_Distributed_Prog/Lab1/Lab1/Lab1/Product.cpp
#include "Product.h"

Product::Product(const std::string& name, int price)
{
	this->name = name;
	this->price = price;
}

void Product::setPrice(int price)
{
	this->price = price;
}

void Product::setName(std::string name)
{
	this->name = name;
}

const int& Product::getPrice()
{
	return this->price;
}

const std::string& Product::getName()
{
	return this->name;
}
